app/src/main/cpp/skyline/kernel/types/KSharedMemory.cpp

```diff
@@ -43,15 +43,33 @@ namespace skyline::kernel::type {
         return guest.ptr;
     }
 
+    void KSharedMemory::Unmap(u8 *ptr, u64 size) {
+        if (!state.process->memory.base.IsInside(ptr) || !state.process->memory.base.IsInside(ptr + size))
+            throw exception("KPrivateMemory allocation isn't inside guest address space: 0x{:X} - 0x{:X}", ptr, ptr + size);
+        if (!util::PageAligned(ptr) || !util::PageAligned(size))
+            throw exception("KSharedMemory mapping isn't page-aligned: 0x{:X} - 0x{:X} (0x{:X})", ptr, ptr + size, size);
+        if (guest.ptr != ptr && guest.size != size)
+            throw exception("Unmapping KSharedMemory partially is not supported: Requested Unmap: 0x{:X} - 0x{:X} (0x{:X}), Current Mapping: 0x{:X} - 0x{:X} (0x{:X})", ptr, ptr + size, size, guest.ptr, guest.ptr + guest.size, guest.size);
+
+        if (mmap(ptr, size, PROT_NONE, MAP_ANONYMOUS | MAP_PRIVATE | MAP_FIXED, 0, 0) == MAP_FAILED)
```

Comment:
no unlikely here? I don't understand our policy on this

Reply:
Same as above, our policy is effectively `[[unlikely]]` on any unlikely path that isn't implicitly marked as such